src/Core/StaticConfigTrait.php

```diff
@@ -80,7 +80,7 @@ public static function setConfig(array|string $key, object|array|null $config =
                 throw new LogicException('If config is null, key must be an array.');
             }
             foreach ($key as $name => $settings) {
-                static::setConfig($name, $settings);
+                static::setConfig((string)$name, $settings);
```

Comment:
I don't think this has any effect unfortunately:

https://3v4l.org/9jv5F

Reply:
Well, it didnt throw TypeError here anymore :)

Reply:
Oh, whoops, I read this code wrong. I was thinking of array keys and not the actual line you were changing.